lib/application/view-models/home/collections_vm.dart

```diff
@@ -0,0 +1,173 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:memo/domain/models/collection.dart';
+import 'package:meta/meta.dart';
+
+final collectionsVM = StateNotifierProvider<CollectionsVM>((_) => CollectionsVMImpl());
+
+enum CollectionsFilter { explore, review }
+const availableFilters = CollectionsFilter.values;
+
+abstract class CollectionsVM extends StateNotifier<CollectionsState> {
+  CollectionsVM(CollectionsState state) : super(state);
+
+  /// Updates the current [state] with [filter]
+  Future<void> updateCollectionsFilter(CollectionsFilter filter);
+}
+
+const mockedCollections = [
+  Collection(
+    id: '1',
+    name: 'Swift Fundamentals I',
+    description: 'Swift fundamentals I description, bla bla bla.',
+    category: 'Swift',
+    tags: ['Linguagem de Programação'],
+    timeSpentInMillis: 10000,
+    memoryStability: 0.3,
+  ),
+  Collection(
+    id: '2',
+    name: 'Swift Fundamentals II',
+    description: 'Swift fundamentals II description, bla bla bla.',
+    category: 'Swift',
+    tags: ['Linguagem de Programação'],
+    timeSpentInMillis: 20000,
+    memoryStability: 0.1,
+  ),
+  Collection(
+    id: '3',
+    name: 'Swift Fundamentals III',
+    description: 'Swift fundamentals III description, bla bla bla.',
+    category: 'Swift',
+    tags: ['Linguagem de Programação'],
+  ),
+  Collection(
+    id: '4',
+    name: 'Git Fundamentals I',
+    description: 'Git fundamentals I description, bla bla bla.',
+    category: 'Git',
+    tags: ['Controle de Versão', 'Linha de Comando'],
+  ),
+  Collection(
+    id: '5',
+    name: 'Git Fundamentals II',
+    description: 'Git fundamentals II description, bla bla bla.',
+    category: 'Git',
+    tags: ['Controle de Versão', 'Linha de Comando'],
+  ),
+];
+
+class CollectionsVMImpl extends CollectionsVM {
+  CollectionsVMImpl() : super(LoadingCollectionsState(availableFilters.first)) {
+    _loadCollections();
+  }
+
+  @override
+  Future<void> updateCollectionsFilter(CollectionsFilter filter) async {
+    state = LoadingCollectionsState(filter);
+
+    // TODO(matuella): attach logic
```

Comment:
Is this a missing TODO or it will be updated in a future task? Could be an issue

Reply:
future task